syntax = "proto3";

package timesync;

service TimeSyncService {
  rpc RequestTimestamp (TimeRequest) returns (TimeResponse);
  rpc CheckHealth (HealthCheckRequest) returns (HealthCheckResponse);
}

message TimeRequest {}

message TimeResponse {
  int64 timestamp_ns = 1;
}

message HealthCheckRequest {}

message HealthCheckResponse {
  bool healthy = 1;          // Server health status
  int64 uptime_seconds = 2;  // Server uptime in seconds
  int64 request_count = 3;   // Total number of requests served
  double avg_response_time_ms = 4; // Average response time in milliseconds
}
